lib/apps/app.js

```diff
@@ -98,24 +113,38 @@ app.get('/:id/edit', async (req, res) => {
 
 // UPDATE
 app.post('/:id', async (req, res) => {
-  const alert = await Alert.get(req.params.id);
-  const newAlert = new Alert(req.body);
+  const browser = await createBrowser();
+  console.log('Browser PID from check is: ' + browser.process().pid);
 
-  // if the search changed, reset price history
-  const isSameSearch = alert.signature === newAlert.signature;
+  try {
+    const lock = new Semaphore(MAX_PAGES);
+    const alert = await Alert.get(req.params.id);
+    const newAlert = new Alert(req.body);
 
-  alert.data.fetchingPrices = true;
+    // if the search changed, reset price history
+    const isSameSearch = alert.signature === newAlert.signature;
 
-  if (isSameSearch) {
-    alert.data.priceHistory = [];
-  }
+    alert.data.fetchingPrices = true;
+
+    // I personally like to see the price history even if I modified the price. Uncomment below for original code which clears it.
+    //  if (isSameSearch) {
+    //    alert.data.priceHistory = [];
+    //  }
 
-  alert.update(req.body);
+    alert.update(req.body);
 
-  res.status(303).location(`/${alert.id}`).end();
+    res.status(303).location(`/${alert.id}`).end();
 
-  await alert.getLatestPrice();
-  alert.save();
+    await alert.getLatestPrice(browser, lock);
+    alert.save();
+  } catch (e) {
+    console.error(e);
+    await browser.close();
+    if (browser && browser.process() != null) browser.process().kill('SIGINT');
+  } finally {
+    await browser.close();
+    if (browser && browser.process() != null) browser.process().kill('SIGINT');
```

Comment:
This looks identical to the `catch` block. Couldn't we just run this code in the `finally` block?

Reply:
I believe you could as the finally will run after the try or catch. I just copied what I had done in check.js, which has the same kind of statement, but both files could be cleaned up.